File: leetcode/1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
// Amir Mukhtarov, mukhtarov.amir.a@gmail.com
#include <iostream> 
#include <vector> 
#include <string> 
#include <algorithm> 
#include <unordered_map> 
#include <map> 
#include <unordered_set> 
#include <set> 
#include <queue> 
#include <stack> 
#include <deque> 
#include <cmath> 
#include <limits> 

using namespace std; 
 
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        int maxSum = 0;

        sort(nums.begin(), nums.end());

        for (int i = 0; i < nums.size() / 2; i++) {
          maxSum = max(maxSum, nums[i] + nums[nums.size() - 1- i]);
        }

        return maxSum;
    }
};


int main(){ 
  vector<int> nums = {3,5,4,2,4,6};
  Solution s;
  cout << s.minPairSum(nums) << endl;

  return 0;
}

File: leetcode/1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.py
# Amir Mukhtarov, mukhtarov.amir.a@gmail.com
from typing import List


class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        max_sum = 0

        # sort from min to max
        nums.sort()

        for i in range(len(nums) // 2):
            max_sum = max(max_sum, nums[i] + nums[-1 - i])

        return max_sum


if __name__ == "__main__":
    nums = [3, 5, 4, 2, 4, 6]

    s = Solution()
    print(s.minPairSum(nums))
